components/timezone/src/ixdtf.rs

```diff
@@ -78,116 +78,214 @@ impl UtcOffset {
 
 // ==== TimeZoneInfo methods and traits ====
 
-impl TimeZoneInfo {
-    fn try_from_ixdtf_record(ixdtf_record: &IxdtfParseRecord) -> Result<Self, ParseError> {
-        match ixdtf_record {
+impl TimeZoneInfo<models::Full> {
+    fn try_offset_only_from_ixdtf_record(
+        ixdtf_record: &IxdtfParseRecord,
+    ) -> Result<Self, ParseError> {
+        let (offset, date, time) = match ixdtf_record {
+            // -0800
             IxdtfParseRecord {
                 offset: Some(offset),
                 tz: None,
+                date: Some(date),
+                time: Some(time),
                 ..
-            } => Self::try_from_utc_offset_record(offset),
+            } => (offset, date, time),
+            // [-0800]
             IxdtfParseRecord {
                 offset: None,
-                tz: Some(time_zone_annotation),
-                date,
-                time,
+                tz:
+                    Some(TimeZoneAnnotation {
+                        tz: TimeZoneRecord::Offset(offset),
+                        ..
+                    }),
+                date: Some(date),
+                time: Some(time),
                 ..
-            } => Self::try_from_time_zone_record(&time_zone_annotation.tz, None, date, time),
+            } => (offset, date, time),
+            // -0800[-0800]
             IxdtfParseRecord {
                 offset: Some(offset),
-                tz: Some(time_zone_annotation),
-                date,
-                time,
+                tz:
+                    Some(TimeZoneAnnotation {
+                        tz: TimeZoneRecord::Offset(offset1),
+                        ..
+                    }),
+                date: Some(date),
+                time: Some(time),
                 ..
             } => {
-                let custom_from_offset = Self::try_from_utc_offset_record(offset)?;
-                Self::try_from_time_zone_record(
-                    &time_zone_annotation.tz,
-                    custom_from_offset.offset,
-                    date,
-                    time,
-                )
+                if offset != offset1 {
+                    return Err(ParseError::InconsistentTimeZoneOffsets);
+                }
+                (offset, date, time)
             }
-            _ => Err(ParseError::MissingTimeZone),
-        }
-    }
-
-    fn try_from_utc_offset_record(record: &UTCOffsetRecord) -> Result<Self, ParseError> {
+            _ => return Err(ParseError::MismatchedTimeZoneFields),
+        };
+        let offset = UtcOffset::try_from_utc_offset_record(offset)?;
+        let iso = DateTime::<Iso>::try_new_iso_datetime(
+            date.year,
+            date.month,
+            date.day,
+            time.hour,
+            time.minute,
+            time.second,
+        )?;
```

Comment:
Nit: `UtcOffset` can construct itself from `OffsetRecord`, but `DateTime` cannot construct itself from `DateRecord`. Seems inconsistent.

Also, maybe create a `Date` and a `Time` instead of a `DateTime`, as that's what we need.

Reply:
it's because `DateTime` is in a different crate and we can't export from-ixdtf-record functions across crate boundaries because ixdtf is less stable than icu_calendar.